infer/schema.go

```diff
@@ -57,15 +57,18 @@ func getAnnotated(t reflect.Type) introspect.Annotator {
 		for k, v := range src.DefaultEnvs {
 			(*dst).DefaultEnvs[k] = v
 		}
+		for k, v := range src.DeprecationMessage {
+			(*dst).DeprecationMessage[k] = v
+		}
 		dst.Token = src.Token
 		dst.Aliases = append(dst.Aliases, src.Aliases...)
-		dst.DeprecationMessage = src.DeprecationMessage
 	}
 
 	ret := introspect.Annotator{
-		Descriptions: map[string]string{},
-		Defaults:     map[string]any{},
-		DefaultEnvs:  map[string][]string{},
+		Descriptions:       map[string]string{},
+		Defaults:           map[string]any{},
+		DefaultEnvs:        map[string][]string{},
+		DeprecationMessage: map[string]string{},
```

Comment:
Nit: this should be plural like `Descriptions`, `Defaults` and `DefaultEnvs`.
```suggestion
		DeprecationMessages: map[string]string{},
```